src/main/java/org/wildfly/security/auth/server/SecurityRealm.java

```diff
@@ -66,6 +70,33 @@ default SupportLevel getEvidenceVerifySupport(String credentialName) throws Real
     }
 
     /**
+     * Handle a realm event.  These events allow the realm to act upon occurrences that are relevant to policy of
+     * the realm; for example, the realm may choose to increase password iteration count on authentication success,
+     * or change the salt of a password after a certain number of authentications.
+     * <p>
+     * The default implementation does nothing.
+     *
+     * @param event the realm event
+     */
+    default void handleRealmEvent(RealmEvent event) {}
```

Comment:
Before we merge all of our realm implementations that wrap other realm implementations e.g. aggregate / privileged will need to have this method added so it passes through.